### PR TITLE
Custom FormatSpec (#439)

### PR DESCRIPTION
**Pull request by polyvertex**

### CODE REVIEW

**Comment by polyvertex** [comment]
Tested on VS2015 only... Obviously.

**Comment by vitaut** [comment]
Thanks for the PR. Mostly looks good but could you default the `Spec` format argument to `FormatSpec` and remove explicit passing of this argument wherever possible to make it a bit cleaner?

**Comment by polyvertex** [comment]
Clean enough?

File: doc/api.rst [hunk 1]
```diff
@@ -120,7 +120,7 @@ custom argument formatter class::
   // A custom argument formatter that formats negative integers as unsigned
   // with the ``x`` format specifier.
   class CustomArgFormatter :
-    public fmt::BasicArgFormatter<CustomArgFormatter, char>  {
+    public fmt::BasicArgFormatter<CustomArgFormatter, char> {
```
Comment (vitaut) [inline doc/api.rst]: nit: extra space

File: test/custom-formatter-test.cc [hunk 2]
```diff
@@ -14,8 +14,8 @@ using fmt::BasicPrintfArgFormatter;
 
 // A custom argument formatter that doesn't print `-` for floating-point values
 // rounded to 0.
-class CustomArgFormatter
-  : public fmt::BasicArgFormatter<CustomArgFormatter, char> {
+class CustomArgFormatter :
+    public fmt::BasicArgFormatter<CustomArgFormatter, char> {
```
Comment (vitaut) [inline test/custom-formatter-test.cc]: Please keep the original formatting here and below.

File: doc/api.rst (hunk 1 above)
Comment (polyvertex) [inline doc/api.rst]: Thought it was just a mistake. Why the extra space?

Comment (vitaut) [inline doc/api.rst]: LOL, my bad. I misread the diff and thought it was added. No need for an extra space of course.

File: test/custom-formatter-test.cc (hunk 2 above)
Comment (polyvertex) [inline test/custom-formatter-test.cc]: Note that I have changed this one to comply to the one below, which I left untouched from master:

```c++
class CustomPrintfArgFormatter :
    public BasicPrintfArgFormatter<CustomPrintfArgFormatter, char> {
```

I'll correct both then

File: doc/api.rst (hunk 1 above)
Comment (polyvertex) [inline doc/api.rst]: (git is so much fun)

**Comment by polyvertex** [comment]
Style changes applied. It got quite noisy here so I can create an all-in-one PR once you've validated the whole thing if you prefer.

**Comment by vitaut** [comment]
No worries, I squashed it via GitHub interface. Thanks for contributing!

**Pull request merged**